code/parse/sexp.cpp

```diff
@@ -12299,6 +12300,43 @@ void sexp_hud_force_static(int n)
 	Sensor_static_forced = is_sexp_true(n);
 }
 
+void sexp_hud_force_emp(int n)
+{
+	bool is_nan, is_nan_forever;
+
+	int intensity = eval_num(n, is_nan, is_nan_forever);
+	if (is_nan || is_nan_forever)
+		return;
+
+	n = CDR(n);
+
+	int duration = eval_num(n, is_nan, is_nan_forever);
+	if (is_nan || is_nan_forever)
+		return;
+
+	n = CDR(n);
+
+	SCP_string text = "";
+
+	if (n >= 0) {
+
+		// assume just text
+		text = CTEXT(n);
+
+		// but use an actual message if one exists
+		for (int i = 0; i < Num_messages; i++) {
+			if (!stricmp(Messages[i].name, text.c_str())) {
+				text = Messages[i].message;
+				break;
+			}
+		}
+
+		text = message_translate_tokens(text.c_str());
+	}
+
+	emp_start_local(i2fl(intensity), i2fl(duration/1000), text);
```

Comment:
The division by 1000 should be done outside the parentheses, after the cast to float.  Inside the parentheses, it's integer division and will be rounded.
```suggestion
	emp_start_local(i2fl(intensity), i2fl(duration)/1000, text);
```

Reply:
done